.github/workflows/deploy-prerelease.yml

```diff
@@ -0,0 +1,234 @@
+          echo "dependency-names=${dependency_names[@]}" >> $GITHUB_OUTPUT
+          echo "pr-numbers=$(sed 's/ /,/' <<< [${pr_numbers[@]}])" >> $GITHUB_OUTPUT
+    
+      - name: Setup Micromamba
+        uses: mamba-org/setup-micromamba@f8b8a1e23a26f60a44c853292711bacfd3eac822 #v1.9.0
+        with:
+          micromamba-version: '1.5.8-0'
+          environment-file: ${{env.ENV_NAME}}
+          environment-name: ${{ steps.get-env-name.outputs.full-name }}
+          generate-run-shell: true
+      
+      # This step is needed to install local conda packages along with their dependencies (https://stackoverflow.com/a/68131606/21024780)
+      - name: Create conda channels
+        shell: micromamba-shell {0}
+        run: |
+          for dependency_repo in ${{env.DEPENDENCY_REPOS}}; do
+            cmd="conda index ${dependency_repo}${{env.CHANNEL_SUFFIX}}"
+            echo "$cmd"
+            eval "$cmd"
+          done
+
+      - name: Install dependencies
+        shell: micromamba-shell {0}
+        run: |
+          channels=$(for dependency_repo in ${{env.DEPENDENCY_REPOS}}; do \
+          echo -n "-c file://${{github.workspace}}/${dependency_repo}${{env.CHANNEL_SUFFIX}} "; done)
+
+          cmd="micromamba install ${{steps.get-dependencies.outputs.dependency-names}} \
+          -c accessnri -c conda-forge -c coecms $channels -y"
+          echo "$cmd"
+          eval "$cmd"
+      
+      - name: Create Pack
+        shell: micromamba-shell {0}
+        run: |
+          conda pack -o ${{ steps.get-env-name.outputs.full-name }}.tar.gz
+
+      - name: Upload Artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{ steps.get-env-name.outputs.full-name }}
+          if-no-files-found: error
+          path: ${{ steps.get-env-name.outputs.full-name }}.tar.gz
+
+  deploy:
+    runs-on: ubuntu-latest
+    needs:
+      - get-deployment-sites
+      - pack
+    strategy:
+      fail-fast: false
+      matrix:
+        deployment-sites: ${{ fromJson(needs.get-deployment-sites.outputs.deployment-sites) }}
+    environment: ${{ matrix.deployment-sites }}
+    permissions:
+      contents: write
+    steps:
+      - uses: actions/download-artifact@v4
+        with:
+          name: ${{ needs.pack.outputs.full-name }}
+
+      - name: Set up SSH
+        uses: access-nri/actions/.github/actions/setup-ssh@main
+        id: ssh
+        with:
+          hosts: |
+            ${{ secrets.HOST_DATA }}
+            ${{ secrets.HOST }}
+          private-key: ${{ secrets.SSH_KEY }}
+
+      - name: Copy to ${{ matrix.deployment-sites }}
+        run: |
+          rsync -v -e 'ssh -i ${{ steps.ssh.outputs.private-key-path }}' \
+            ${{ needs.pack.outputs.full-name }}.tar.gz \
+            ${{ secrets.USER }}@${{ secrets.HOST_DATA }}:${{ vars.PACK_DIR }}
+
+      - name: Deploy to ${{ matrix.deployment-sites }}
+        env:
+          DEPLOYMENT_DIR: ${{ vars.ENVIRONMENT_DIR }}/${{ env.VERSION }}
```

Comment:
I think the pre-release should have it's own environment and should go under `/g/data/vk83/prerelease`

See https://github.com/ACCESS-NRI/ACCESS-OM2/settings/environments/2174012585/edit as an example.

Reply:
I don't have access to ACCESS-OM2 settings, but I checked the [ci.yml workflow](https://github.com/ACCESS-NRI/build-cd/blob/main/.github/workflows/ci.yml) referenced in it.

Ok, so the pre-release modules should go to `g/data/vk83/prerelease/modules` instead of `g/data/vk83/modules`.
Makes sense. 
I'll change the ENVIRONMENT_DIR for pre-release and the module files accordingly

Reply:
Refactored in [ace9298cb0781bc191738b818419428925f39eaf](https://github.com/ACCESS-NRI/access-ram-condaenv/pull/9/commits/ace9298cb0781bc191738b818419428925f39eaf). 

I also reset most of the Environment Variables (`vars`) for the [Gadi GitHub environment](https://github.com/ACCESS-NRI/access-ram-condaenv/settings/environments/3687957553/edit), so the workflow could be better generalized.
The portions relative to the deployment on Gadi for the [release](https://github.com/ACCESS-NRI/access-ram-condaenv/blob/ace9298cb0781bc191738b818419428925f39eaf/.github/workflows/deploy-release.yml#L171-L198), [prerelease](https://github.com/ACCESS-NRI/access-ram-condaenv/blob/ace9298cb0781bc191738b818419428925f39eaf/.github/workflows/deploy-prerelease.yml#L208-L235) and [modulefiles](https://github.com/ACCESS-NRI/access-ram-condaenv/blob/ace9298cb0781bc191738b818419428925f39eaf/.github/workflows/update-module.yml#L38-L94) workflows could now be applied to any module we need to deploy (for example it would work as well for `payu` and any other future module we would release in `vk83` (most likely we are going to maintain the naming format as `<module-name>/version`.

In the future, a general action (or probably 2 actions, one for the modulefiles, one for the release/prerelease modules) could be created for this (maybe under [ACCESS-NRI/actions](https://github.com/ACCESS-NRI/actions)) so we can reuse it in any repo that needs to deploy modules.
For the actions to work, the following Environment Variables and secrets should also be set for every deployment environment (I show the example for the _Gadi_ environment Variables in this repo):
**Environment Variables**
- ROOT_DIR (`/g/data/vk83`)
- ROOT_DIR_PRERELEASE (`/g/data/vk83/prerelease`)
- MODULES_DIR (`modules`)
- APPS_DIR (`apps`)
- PACK_DIR (`/scratch/tm70/tm70_ci/access-ram-condaenv`)
- MODULE_NAME (`conda/access-ram`)

**Environment Secrets**
- HOST
- HOST_DATA
- SSH_KEY
- USER

Potentially, the only ones that vary across different repos are only `PACK_DIR` and `MODULE_NAME`. However, organization-level environments cannot be set (yet) on GitHub, so I think we might have to manually set all these variables for every repo (as we are doing right now anyway).